src/riscv/lib/src/machine_state/instruction.rs

```diff
@@ -1680,10 +1680,12 @@ impl Args {
 
     /// SAFETY: This function must only be called on an `Args` belonging
     /// to the same OpCode as the OpCode used to derive this function.
-    unsafe fn run_jal<I: ICB>(&self, icb: &mut I) -> IcbFnResult<I> {
+    unsafe fn run_jump_and_link_pc<I: ICB>(&self, icb: &mut I) -> IcbFnResult<I> {
+        // link the return address to the program counter
         let rd = unsafe { self.rd.nzx };
-        let addr = branching::run_jal(icb, self.imm, rd, self.width);
-        let pcu = ProgramCounterUpdate::Set(addr);
+        branching::run_add_immediate_to_pc(icb, self.width as i64, rd);
```

Comment:
I'm a little confused by this. Why are we adding an immediate to the PC here?

Reply:
I see. The name is quite confusing in this context. It makes sense as an instruction, though.